testutil/keeper/unit_test_helpers.go

```diff
@@ -101,4 +103,53 @@ func GetIBCQueryKeeperAndCtx(t *testing.T, params InMemKeeperParams) (
 	ctrl := gomock.NewController(t)
 	mocks := NewMockedKeepers(ctrl)
 	return NewInMemIBCQueryKeeper(params, mocks), params.Ctx, ctrl, mocks
+}
+
+// Returns an in-memory provider keeper, context, controller, and mocks after query data is stored in the store
+func GetSubmitCrossChainQueryKeeperAndCtx(IBCQueryKeeper ibcquerykeeper.Keeper, 
+	ctx sdk.Context, mocks MockedKeepers) (
+	ibcquerykeeper.Keeper, sdk.Context, MockedKeepers, *types.MsgSubmitCrossChainQueryResponse) {
```

Comment:
This function's works are so many I think
1. create CrossChainQuery
2. Set Mock
3. Submit query to keeper
4. return keeper

If one function does many works, it's hard to understand and diffcult to change function

I think spliting this function makes code more better